client/dfclient/src/http_client.cpp

```diff
@@ -0,0 +1,82 @@
+#include "http_client.hpp"
+
+#ifdef __EMSCRIPTEN__
+// TODO: write an emscripten version
+int http::MatchmakerGet(std::string& ret)
+{
+	return -1;
+}
+#else // __EMSCRIPTEN__
+
+#include <boost/beast/core.hpp>
+#include <boost/beast/http.hpp>
+#include <boost/beast/version.hpp>
+#include <boost/asio/connect.hpp>
+#include <boost/asio/ip/tcp.hpp>
+#include <cstdlib>
+#include <iostream>
+#include <string>
+
+namespace beast = boost::beast;     // from <boost/beast.hpp>
+namespace bhttp = beast::http;      // from <boost/beast/http.hpp>
+namespace net = boost::asio;        // from <boost/asio.hpp>
+using tcp = net::ip::tcp;           // from <boost/asio/ip/tcp.hpp>
+
+static int throwingMatchmakerGet(std::string address, std::string port, std::string& ret)
+{
+	// The io_context is required for all I/O
+	net::io_context ioc;
+
+	// These objects perform our I/O
+	tcp::resolver resolver(ioc);
+	beast::tcp_stream stream(ioc);
+
+	// Look up the domain name
+	auto const results = resolver.resolve(address, port);
+
+	// Make the connection on the IP address we get from a lookup
+	stream.connect(results);
+
+	// Set up an HTTP GET request message
+	bhttp::request<bhttp::string_body> req{bhttp::verb::get, "/matchmake", 10};
+	req.set(bhttp::field::host, address);
+	req.set(bhttp::field::user_agent, BOOST_BEAST_VERSION_STRING);
+
+	// Send the HTTP request to the remote host
+	bhttp::write(stream, req);
+
+	// This buffer is used for reading and must be persisted
```

Comment:
This comment is a little imprecise. Persisted, but for how long? What does "persisted" mean here?

Reply:
Actually read the comments copied from an example. Removed the "persisted" part. Everything else checks out